### PR TITLE
fixed Redis subscriber adapter, replace all occurrence of `wildcardOne` symbol ('+') in topic

### PR DESCRIPTION
**Pull request by emilianobonassi**
Fixes https://github.com/mcollina/mqemitter-redis/issues/5

### CODE REVIEW

**Comment by emilianobonassi** [comment]
Sure

**Comment by emilianobonassi** [comment]
Done

**Comment by emilianobonassi** [comment]
@mcollina , `safe-buffer` module causes some build to fail (0.12, 0.4), may you check?

**Comment by mcollina** [comment]
Can you add it as a devDependency? that's the easiest fix.

**Comment by emilianobonassi** [comment]
@mcollina , oks.

File: mqemitter-redis.js [hunk 1]
```diff
@@ -91,7 +91,7 @@ MQEmitterRedis.prototype.close = function (cb) {
 }
 
 MQEmitterRedis.prototype._subTopic = function (topic) {
-  return topic.replace(this._opts.wildcardOne, '*')
+  return topic.replace(new RegExp(this._opts.wildcardOne.replace(/([/,!\\^${}[\]().*+?|<>\-&])/g, '\\$&'), 'g'), '*')
```
Comment (mcollina) [inline mqemitter-redis.js]: can you avoid instantiating a new RegExp for every call to `_subTopic`? This is going to be slow.

Comment (emilianobonassi) [reply mqemitter-redis.js]: Thanks for the review. Yes, of course. I am aware of the performance issue.  I did in the first version (in the constructor/lazy init in the method) but I've noticed that you can change the symbol after instantation ( a unit test failed ). I will intercept the setter call and replace the regex with the new one (if the symbol will change obviously). I will do it at the beginning of the next week.

Comment (emilianobonassi) [reply mqemitter-redis.js]: I remembered a false fact 

>  I've noticed that you can change the symbol after instantation ( a unit test failed )

Now RegExp is instantiated in the constructor and tests pass without any issue.

**Comment by mcollina** [comment]
Thanks!

**Pull request merged**